Sourcery/Configuration.swift

```diff
@@ -80,13 +96,20 @@ enum Source {
     case projects([Project])
     case sources(Paths)
 
-    init(dict: [String: Any], relativePath: Path) {
+    init(dict: [String: Any], relativePath: Path) throws {
         if let projects = dict["project"] as? [[String: Any]] {
-            self = .projects(projects.flatMap({ Project.init(dict: $0, relativePath: relativePath) }))
+            guard !projects.isEmpty else { throw Configuration.Error.invalidSources(message: "No projects provided.") }
+            self = try .projects(projects.map({ try Project(dict: $0, relativePath: relativePath) }))
         } else if let project = dict["project"] as? [String: Any] {
-            self = .projects([Project(dict: project, relativePath: relativePath)].flatMap({ $0 }))
+            self = try .projects([Project(dict: project, relativePath: relativePath)])
```

Comment:
I think just popping error here is fine, though yes, we can catch it and wrap in invalid sources error.